network/p2p/gossip/bloom_test.go

```diff
@@ -56,43 +57,50 @@ func TestBloomFilterRefresh(t *testing.T) {
 				{id: ids.ID{2}},
 			},
 		},
+		{
+			name:                           "multiple refresh",
+			minTargetElements:              1,
+			targetFalsePositiveProbability: 0.01,
+			resetFalsePositiveProbability:  0.0000000000000001, // maxCount = 1
+			resetCount:                     2,
+			add: []*testTx{
+				{id: ids.ID{0}},
+				{id: ids.ID{1}},
+				{id: ids.ID{2}},
+				{id: ids.ID{3}},
+				{id: ids.ID{4}},
+			},
+			expected: []*testTx{
+				{id: ids.ID{4}},
+			},
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			require := require.New(t)
-			numHashes, numEntries := bloom.OptimalParameters(
-				tt.minTargetElements,
-				tt.targetFalsePositiveProbability,
-			)
-			b, err := bloom.New(numHashes, numEntries)
+			bloom, err := NewBloomFilter(prometheus.NewRegistry(), tt.minTargetElements, tt.targetFalsePositiveProbability, tt.resetFalsePositiveProbability)
 			require.NoError(err)
-			bloom := BloomFilter{
-				bloom:                          b,
-				maxCount:                       bloom.EstimateCount(numHashes, numEntries, tt.resetFalsePositiveProbability),
-				minTargetElements:              tt.minTargetElements,
-				targetFalsePositiveProbability: tt.targetFalsePositiveProbability,
-				resetFalsePositiveProbability:  tt.resetFalsePositiveProbability,
-			}
 
-			var didReset bool
+			var resetCount uint64
 			for _, item := range tt.add {
 				bloomBytes, saltBytes := bloom.Marshal()
 				initialBloomBytes := slices.Clone(bloomBytes)
 				initialSaltBytes := slices.Clone(saltBytes)
 
-				reset, err := ResetBloomFilterIfNeeded(&bloom, len(tt.add))
+				reset, err := ResetBloomFilterIfNeeded(bloom, len(tt.add))
 				require.NoError(err)
 				if reset {
-					didReset = reset
+					resetCount++
 				}
 				bloom.Add(item)
 
 				require.Equal(initialBloomBytes, bloomBytes)
 				require.Equal(initialSaltBytes, saltBytes)
 			}
 
-			require.Equal(tt.reset, didReset)
+			require.Equal(tt.resetCount, resetCount)
+			require.Equal(float64(tt.resetCount), testutil.ToFloat64(bloom.metrics.resetCount))
```

Comment:
How did I never know about `"github.com/prometheus/client_golang/prometheus/testutil"`?? 🤯

Reply:
cuz we never added UTs for metrics? (or was it we never added them cuz we did not know this???)